internal/driver/glfw/animation.go

```diff
@@ -0,0 +1,77 @@
+package glfw
+
+import (
+	"time"
+
+	"fyne.io/fyne"
+)
+
+type anim struct {
+	a     *fyne.Animation
+	start time.Time
+	end   time.Time
+}
+
+func (d *gLDriver) runAnimations() {
+	draw := time.NewTicker(time.Second / 60)
+
+	go func() {
+		done := false
+		for !done {
+
+			<-draw.C
+			d.animationMutex.Lock()
+			oldList := d.animations
+			d.animations = nil // clear the list so we can append any new ones after processing
+			d.animationMutex.Unlock()
+			var newList []*anim
+			for _, a := range oldList {
+				if d.tickAnimation(a) {
+					newList = append(newList, a)
+				}
+			}
+			d.animationMutex.Lock()
+			d.animations = append(newList, d.animations...)
+			done = len(newList) == 0
```

Comment:
Is it possible newList is empty, but another animation was added in between? Or would that be executed by another goroutine? In which case an animation could end up being ticked by multiple threads
```suggestion
			done = len(d.animations) == 0
```

Reply:
We only have 1 thread running - this loop is handled by the driver.
New animations cannot be added to this list during a tick because of the lock - this loop takes a snapshot of `d.animations` and works with it. Line 34 could probably just be `d.animations = newList` now that I think about it - it just left us open to letting things drop through if the locking changed.

Reply:
> New animations cannot be added to this list during a tick because of the lock

I disagree, the lock is unlocked on line 26 and relocked on line 33, so an animation could be added between them.

Reply:
Hah, I missed that. I really should just work on one feature at a time